### ISSUE TITLE
Can't build hello_world example with 1.44.0-nightly

### ISSUE THREAD

**Issue by shaneutt** [SergioBenitez/Rocket#1258]
# The version of Rocket you're using. Ensure it's the latest, if possible

`master`

# The operating system (distribution and version) where the issue occurs

Arch Linux 2020.03.01 Kernel 5.5.6

# Report

It had been a while since I used Rocket last, and I decided to go through the [quickstart](https://rocket.rs/v0.4/guide/quickstart/) and run the `hello_world` example to start getting re-acquainted.

When trying to compiled the `hello_world` example, I received the following:

```
$ cargo run
   Compiling time v0.2.8
error: expected an item keyword
   --> /home/shane/.cargo/registry/src/github.com-1ecc6299db9ec823/time-0.2.8/src/utc_offset.rs:366:13
    |
366 |             let tm = timestamp_to_tm(datetime.timestamp())?;
    |             ^^^

error: aborting due to previous error

error: could not compile `time`.

To learn more, run the command again with --verbose.
```

This compilation issue appears to be fixed upstream in `v0.2.9` of [time](https://crates.io/crates/time).

**Comment by messense** [comment]
I think you can fix this by running `cargo update`

**Comment by shaneutt** [comment]
> I think you can fix this by running `cargo update`

Thank you, but #1259 seems to resolve the problem without having to do anything different from the quickstart guide, so I would like to pursue that if the maintainers are interested.

**Comment by SergioBenitez** [comment]
Exactly as @messense suggests.

**Comment by shaneutt** [comment]
Ok your choice of course. I just wanted to point out that the effect of not making some improvement here as I understand it seems to be that the quickstart may not necessarily compile for new users as it is currently written.

**Comment by jebrosen** [comment]
I tried it and as far as I can tell the quickstart *does* work as currently written: `cargo` will automatically create a new `Cargo.lock` file with `time 0.2.9` because that's the latest semver-compatibile release. Did you perhaps previously try to compile on March 12th, when `time 0.2.8` was released but `time 0.2.9` was not out yet? That would lead to `time 0.2.8` being placed in `Cargo.lock` which would be reused the next time you tried.

Note that your PR would have no impact on the quickstart instructions anyway, because the `v0.4` branch of rocket uses `time 0.1`. It's a different dependency somewhere upstream that would have brought in `time 0.2`.

**Comment by SergioBenitez** [comment]
@shaneutt You were right: the bump seems to be required, for some reason. Cargo simply doesn't do the right thing, at least on Linux in my particular environment. In particular it continued to choose `time` `0.2.8`, even though `0.2.9` was available. Manually bumping the version, while logically making no difference, results in Cargo picking the right version. My apologies for hastily closing your issue and pull request. Thank you bringing this to our attention.
